### PR TITLE
Bug 1797997 - Switch taskcluster policy to `public_restricted`

### PR DESCRIPTION
**Pull request by JohanLorenzo**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
bors try

**Review by ahal — CHANGES_REQUESTED** [review]
Wait, where does this change the pr policy? It doesn't touch `.taskcluster.yml`, or am I missing something?

**Review by JohanLorenzo — COMMENTED** [review]
Sorry for the confusion, I actually had to create this draft PR to get the parameters files for a `bors try` graph. I actually had to update all the test param files because of the migration: all commits had their hashes changed.

After these changes, I ran `taskgraph target -p taskcluster/test/params/main-repo-pull-request.yml --diff HEAD~1` to see what tasks the trusted pull requests now get. Here's the diff:

```diff
--- target_task_set@bug-1797997~1
+++ target_task_set@bug-1797997
@@ -106,23 +106,35 @@
 build-support-test-fakes
 build-support-test-libstate
 build-support-utils
 build-support-webextensions
 build-tooling-detekt
 build-tooling-fetch-tests
 build-tooling-glean-gradle
 build-tooling-lint
 build-tooling-nimbus-gradle
 build-ui-autocomplete
 build-ui-colors
 build-ui-fonts
 build-ui-icons
 build-ui-tabcounter
 build-ui-widgets
 complete-pr
 complete-pr-1
 complete-pr-2
 lint-compare-locales
 lint-detektAndKtlint
+test-android-feature-containers
+test-android-feature-downloads
+test-android-feature-logins
+test-android-feature-prompts
+test-android-feature-pwa
+test-android-feature-share
+test-android-feature-sitepermissions
+test-android-feature-top-sites
+test-android-support-ktx
+test-ui-browser
+test-ui-glean
+test-unit-browser-engine-gecko-nightly
 toolchain-linux64-android-gradle-dependencies
 toolchain-linux64-android-sdk-linux-repack
```

I checked the untrusted PRs and the set of tasks remained the same.

**Comment by JohanLorenzo** [comment]
Oh, I busted the decision task because [D161278](https://phabricator.services.mozilla.com/D161278) hasn't landed yet. I'll add the UI tests jobs after this PR then that phabricator revision are merged. It'll be in a follow-up PR.

**Review by ahal — CHANGES_REQUESTED** [review]
You'll need to replace any reference to `tasks_for == "github-pull-request"` in `.taskcluster.yml` with:
```
tasks_for[:19] == "github-pull-request"
```

(unless the logic *should* only apply to a trusted PR)

Or alternatively you could create an `isPullRequest` variable ([e.g](https://github.com/taskcluster/taskgraph/blob/3481bb676219541dedbd9ef1efd27e0e0875e1eb/.taskcluster.yml#L118))

File: .taskcluster.yml [hunk 1]
```diff
@@ -2,341 +2,347 @@
...
-                  then: 'skaspari+mozlando@mozilla.com'   # It must match what's in bors.toml
+                  $if: 'tasks_for == "cron"'
+                  then: '${ownTaskId}'
+
+          pullRequestAction:
+              $if: 'isPullRequest'
+              then: ${event.action}
+              else: 'UNDEFINED'
+      in:
+          $let:
```
Comment (JohanLorenzo) [inline .taskcluster.yml:26]: I added one level of indentation reuse `isPullRequest` wherever possible.

**Review by JohanLorenzo — COMMENTED** [review]
Oh yeah, thanks for reminding me that! I had seen it in your original PR but that slipped my mind a few hours later 😅

File: .taskcluster.yml [hunk 2]
```diff
@@ -2,341 +2,346 @@
...
                                 then:
-                                    - 'assume:repo:github.com/${repoFullName}:pull-request'
+                                    - 'assume:repo:github.com/${repoFullName}:branch:${short_head_ref}'
                                 else:
-                                    $if: 'tasks_for == "action"'
+                                    $if: 'isPullRequest'
                                     then:
-                                        # when all actions are hooks, we can calculate this directly rather than using a variable
-                                        - '${action.repo_scope}'
+                                        - 'assume:repo:github.com/${repoFullName}:${tasks_for[7:]}'
```
Comment (JohanLorenzo) [inline .taskcluster.yml:220]: I changed this line to support both `github-pull-request` and `github-pull-request-untrusted`, per https://github.com/taskcluster/taskgraph/blob/3481bb676219541dedbd9ef1efd27e0e0875e1eb/.taskcluster.yml#L222

**Review by ahal — APPROVED** [review]
Thanks, this looks good to me!

**Comment by ahal** [comment]
Completely tangential, but it's occurred to me on several occasions that we should totally standardize on two space indent in these files.. I know as Python devs we like our 4 spaces, but in this case I think it'll help a bunch with readability.

**Pull request merged**